gadget-container/entrypoint.sh

```diff
@@ -95,10 +95,21 @@ if [ "$HOOK_MODE" = "crio" ] ; then
     cp /opt/hooks/oci/$i /host/opt/hooks/oci/
   done
 
-  echo "Installing OCI hooks configuration in /usr/share/containers/oci/hooks.d"
+  echo "Installing OCI hooks configuration in /etc/containers/oci/hooks.d and /usr/share/containers/oci/hooks.d"
+  mkdir -p /host/etc/containers/oci/hooks.d
+  cp /opt/hooks/crio/gadget-prestart.json /host/etc/containers/oci/hooks.d/ 2>/dev/null || true
+  cp /opt/hooks/crio/gadget-poststop.json /host/etc/containers/oci/hooks.d/ 2>/dev/null || true
   mkdir -p /host/usr/share/containers/oci/hooks.d
-  cp /opt/hooks/crio/gadget-prestart.json /host/usr/share/containers/oci/hooks.d/gadget-prestart.json
-  cp /opt/hooks/crio/gadget-poststop.json /host/usr/share/containers/oci/hooks.d/gadget-poststop.json
+  cp /opt/hooks/crio/gadget-prestart.json /host/usr/share/containers/oci/hooks.d/ 2>/dev/null || true
+  cp /opt/hooks/crio/gadget-poststop.json /host/usr/share/containers/oci/hooks.d/ 2>/dev/null || true
+
+  if ! ls /host/etc/containers/oci/hooks.d/gadget-{prestart,poststop}.json > /dev/null 2>&1 && \
+     ! ls /host/usr/share/containers/oci/hooks.d/gadget-{prestart,poststop}.json > /dev/null 2>&1
+  then
+    echo "Couldn't install OCI hooks configuration. Clean up files installed so far before exiting" >&2
+    ./cleanup.sh
+    exit 1
```

Comment:
I don't think we should have a hard failure here. Inspektor Gadget will able to work (through the pod informer fallback) if  other hooks fail. I think just a warning here is enough.

Reply:
You are right.